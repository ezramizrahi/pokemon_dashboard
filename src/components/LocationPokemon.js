import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { Label, Loader } from 'semantic-ui-react';

const LocationPokemon = ( props ) => {
  const url = props.url;
  const [pokemon, setPokemon] = useState([]);

  useEffect(() => {
    const getPokemon = async (url) => {
      try {
        const response = await axios.get(url);
        setPokemon(response);
      } catch (err) {
        console.log(err);
      }
    };
    getPokemon(url);
  }, [url])

  return (
    <div>
      {(pokemon && pokemon?.data)
        ?
        <Label as='a' color='blue' size='mini' image>
          <img alt={pokemon.data.name} src={pokemon.data.sprites.front_shiny} />
          {pokemon.data.name}
          <Label.Detail>
            Type(s): {pokemon.data.types.map((t, index) =>
              <span key={index}>{t.type.name} </span>
            )}
          </Label.Detail>
        </Label>
        :
        <Loader size='mini'>Fetching Location Pok√©mon Info...</Loader>
      }
    </div>
  )
};

export default LocationPokemon;
