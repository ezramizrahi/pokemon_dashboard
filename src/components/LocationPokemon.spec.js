import React from 'react';
import renderer from 'react-test-renderer';
import LocationPokemon from './LocationPokemon';

describe('LocationPokemon Component', () => {
  it('renders loading message if props are empty', () => {
    const props = {};
    const tree = renderer
      .create(<LocationPokemon {...props}></LocationPokemon>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
