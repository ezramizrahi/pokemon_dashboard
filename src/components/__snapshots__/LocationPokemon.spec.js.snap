// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPokemon Component renders loading message if props is empty 1`] = `
<div>
  <div
    className="ui mini text loader"
  >
    Fetching Pok√©mon Location Info...
  </div>
</div>
`;
